spec/api/comment_thread_spec.rb

```diff
@@ -668,7 +667,7 @@ def test_unicode_data(text)
       let :default_params do
         {body: "new comment", course_id: "1", user_id: User.first.id}
       end
-      it "create a comment to the comment thread" do
+      it "create a comment to the comment thread and marks thread as read for user" do
```

Comment:
@robrap Currently, `ThreadPresenter` takes in optional `user` parameter in order to calculate thread's `read` status. If not passed; it defaults `read=false` and uses all `comments_count` as `unread_comment_count`; which, may be, are incorrect values.

There are APIs which are not expecting `user_id` input param.
1. thread PUT
2. thread GET (for Web, mobile discussions API passes in `user_id`)
This is where I do not like the idea that we are keeping two separate parallel implementations of same module; because it creates mess in terms of unnecessary checks in forums code.

In order to fix what you have proposed; we need to update these endpoints to always return correct value of `read`.
I propose; we create a new JIRA dev ticket to fix this and introduce a mandatory `user_id` param for these endpoints for both mobile and web implementations.